clang/lib/CIR/CodeGen/CIRGenBuilder.h

```diff
@@ -33,6 +33,11 @@ class CIRGenBuilderTy : public cir::CIRBaseBuilderTy {
       llvm_unreachable("NYI: PPC double-double format for long double");
     llvm_unreachable("Unsupported format for long double");
   }
+
+  bool isSized(mlir::Type ty) {
+    return mlir::isa<cir::PointerType, cir::ArrayType, cir::BoolType,
```

Comment:
I think this was better with the assert for other types. This function should handle all types that can occur in the current form of CIR. As we add new types, this function should be updated to indicate if they are sized or not. So, this isn't quite like other `errorNYI` cases. An assert here is reasonable as a permanent part of the code.

Reply:
Yes i will update it, and i think no need for NYI or other assert when calling it because in call cases we will not need to recover